### ISSUE TITLE
Need option to turn off "Reply" tag

### ISSUE THREAD

**Issue by scovel** [borkke/opentracing-csharp-mongo#12]
For some operations, find, the reply from mongo can be quite large, and can also be sensitive.  I need an option to turn off adding the "reply" tag.  

`activeScope.SetTag($"{MongoDbPrefix}reply", @event.Reply.ToString());`

when debugging, or test environments, adding the complete reply is acceptable, but in a production release it could be a performance issue, as well as a security issue.

Thanks,

Sean

**Comment by borkke** [comment]
Thanks for pointing this out. Probably also makes sense to do the same for the request objects as well. 

[This](https://github.com/borkke/opentracing-csharp-mongo/blob/master/src/OpenTracing.Contrib.Mongo/Tracer/DefaultMongoEventListener.cs#L77) will dump whatever is in the request payload
```bash
....   
    .WithTag(Tags.DbStatement, @event.Command.ToString())
```

**Comment by borkke** [comment]
@scovel will you please let me know if this is something you were looking for?

```c#
var mongoClient = new TracingMongoClient("<connection string>", options =>
{
    ....
    options.MasskedEvents = new[] {"insert"}
});
```

This would allow controlling which events will be masked, and under the hood, it will just replace "command" and "reply" payloads with `******`

**Comment by scovel** [comment]
in my local repo I implemented the following.  Not very elegant, but effective...

```
        private readonly bool _UseReplyTag;

        public MongoEventListener(ITracer tracer, TracingOptions options)
        {
            _tracer = tracer;
            _eventFilter = new EventFilter(options.WhitelistedEvents);            
            _spanCache = new ConcurrentDictionary<int, ISpan>();
            _UseReplyTag = options.UseReplyTag;
        }


        public void SuccessEventHandler(CommandSucceededEvent @event)
        {
            if (!_eventFilter.IsApproved(@event.CommandName))
                return;

            if (_spanCache.TryRemove(@event.RequestId, out var activeScope))
            {
                if (_UseReplyTag)
                {
                    activeScope.SetTag($"{MongoDbPrefix}reply", @event.Reply.ToString());
                }

                activeScope.Finish();
            }
        }
```

**Comment by borkke** [comment]
This would work for sure. The only thing that is missing is `command` which might also contain sensitive information.

I was experimenting locally... Let me see if I can get it out a little bit later today.

**Comment by borkke** [comment]
Can you please try “0.2.3-beta.3”

**Comment by scovel** [comment]
I tested the beta.  It is masking in the trace.  Is there a way where I can mask just the reply, but still keep the command?  I want to know what the "find" command was, I just don't want to see the "reply" in the logs.  Would it be possible to do something like this?

                        options.MasskedEvents = new[] {"insert", "update", "reply"};

And another note "MasskedEvents" should be "MaskedEvents".  One too many "s" in Mask.

Thanks,

Sean

**Comment by borkke** [comment]
Not at the moment but that should not be that hard. Let me see if I can get that today.

**Comment by borkke** [comment]
@scovel Sorry to keep you waiting. The reason why we can't do it similar to your example is that `insert` and `update` are "events" and `reply` is just a field that contains information.

I'll catch up with readme but in meantime, this is the snippet that should work. You can also check out the [test](https://github.com/borkke/opentracing-csharp-mongo/blob/master/test/OpenTracing.Contrib.Mongo.Test/TracingMongoClient/ShouldCreate.cs#L201) here.
```c#
//mask all "insert" events
optiosns.MaskedEvents = new string[] { "insert" };

//mask only "mongodb.reply" field for insert events
optiosns.MaskedFields = new string[] { "mongodb.reply" };
```

**Comment by borkke** [comment]
And there is another beta `0.2.3-beta.5` if you can give it a try, please.

**Comment by scovel** [comment]
Sorry I took so long to respond.  Yes, that is even better than I was hoping for.  I think you can go ahead and release it.  Thanks for the fast support.

**Comment by borkke** [comment]
@scovel Thanks again for a good idea. The latest version is released as `0.3.1`. I hope it works out well for you. 

Closing this now.

**Comment by PavelStefanov** [comment]
Hey guys!

Thanks for the helpful discussion! 

Can I handle however all events for masking? 
My case: I need to mask some fields for all events. If I understand correctly now I must specify each event.  

I can expect something like this:
``` csharp
optiosns.MaskedEvents = new string[] { "*" };
optiosns.MaskedFields = new string[] { "mongodb.reply" };
```
